Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -561,6 +646,204 @@ def use_homebrew_curl?(formula_or_cask, url)
       homebrew_curl_root_domains.include?(url_root_domain)
     end
 
+    # Identifies the latest version of the resource in a given Formulae/Casks and returns a Hash containing
+    # the version information for a resource. Returns a nil value if a latest version
+    # couldn't be found for a given resource.
+    sig {
+      params(
+        resource:  T.any(Resource),
+        json:      T::Boolean,
+        full_name: T::Boolean,
+        verbose:   T::Boolean,
+        debug:     T::Boolean,
+      ).returns(T.nilable(Hash))
+    }
+    def resource_version(
+      resource,
+      json: false,
+      full_name: false,
+      verbose: false,
+      debug: false
+    )
+      has_livecheckable = resource.livecheckable?
+
+      if debug
+        puts "\n\n"
+        puts "Resource:          #{resource_name(resource, full_name: full_name)}"
+        puts "Livecheckable?:    #{has_livecheckable ? "Yes" : "No"}"
+      end
+
+      resource_version_info = {
+        resource: resource_name(resource, full_name: full_name),
+        version:  {
+          current: resource.version,
+        },
+      }
+
+      # Check resources with livecheck block (will be updated in the future)
+      if has_livecheckable
```

Comment:
As discussed earlier on Slack, we should check _all_ resources, not just the ones with `livecheck` blocks.

Reply:
Done ✅